docs/proposals/fleetapiserver/fleetapiserver.md

```diff
@@ -0,0 +1,239 @@
+- "@yinzhang-zen"
+- "@chaunceyjiang"
+approvers:
+- "@kevin-wangzefeng"
+- "@RainbowMango"
+
+creation-date: 2023-11-23
+
+---
+
+# FleetAPIServer -- provide a unified view of resources across multiple clusters
+
+## Summary
+
+In order to distinguish between users' usage scenarios, we introduce the concept of federation. When users do not enable federation, they only use FleetAPIServer to provide a unified view of multi-cluster resource; when users enable federation, Karmada will be deployed to increase the capabilities provided by Karmada.
+
+FleetAPIServer meeting the management needs of both non-federated and federated scenarios, making it convenient for users to manage multiple clusters in a unified manner.
+
+## Motivation
+
+Karmada uses kube-apiserver as the API access entry point, which can provide users with a smooth transition from single-cluster to multi-cluster experience. However, there are still some confusion and inconvenience for users in the actual usage process.
+
+When users use Karmada for resource operations, operation and maintenance, or secondary development, they need to switch kubeconfig context. Different contexts correspond to different targets, such as:
+
+- Use this server address to query Karmada control plane resources: https://172.18.0.2:5443
+- Use this server address to query member1 cluster resources: https://172.18.0.2:5443/apis/cluster.karmada.io/v1alpha1/clusters/member1/proxy
+- Use this server address to query member2 cluster resources: https://172.18.0.2:5443/apis/cluster.karmada.io/v1alpha1/clusters/member2/proxy
+
+For some users, switching the above context requires awareness of cluster information. For existing businesses, certain modification are needed. For developers and operation personal, learning new knowledge is required.
+
+In order to accommodate these users with historical burdens, Karmada needs to take compatibility with Kubernetes single-cluster operation experience as a starting point and further provide smooth migration best practices from single-cluster architecture to multi-cluster architecture.
+
+### Goals
+
+- Add FleetAPIServer to provide users with a unified view of multi-cluster resources.
+- The FleetAPIServer component is pluggable, and not using this component will not affect users' normal use of Karmada functions.
+
+### Non-Goals
+
+- Do not consider performance issues for now.
+- Do not consider the issue of extensive support for different version member cluster APIs at the moment.
+
+## Proposal
+
+From the user's perspective, it looks like this:
+
+![user access view](./statics/user-access-view.png)
+
+Users can directly access FleetAPIServer to search for or operate on all clusters, including resources on Karmada. They can also specify the clusterspace parameter to search for or operate on resources in a specific cluster.
+
+The architecture diagram of FleetAPIServer is given below.
+
+![fleetapiserver architecture](./statics/fleetapiserver-architecture.png)
```

Comment:
Please clarify whether it's required that fleet-apiserver and karmada-apiserver use the same etcd, as it would result in different deployment constraints.
And since the fleet-apiserver and karmada-apiserver are using the same etcd, it's better to differentiate between write and read-only api calls